### PR TITLE
soletrader correspondance text and postcode validation changes

### PR DESCRIPTION
**Pull request by sardhani-ch**

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressManualController.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@ import * as config from "../../../config";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
     req.session.user = req.session.user || {};
-    res.render(config.SOLE_TRADER_MANUAL_CORRESPONDANCE_ADDRESS, {
-        title: "What is your correspondance address?",
-        previousPage: "/sole-trader/correspondance-address--lookup",
+    res.render(config.SOLE_TRADER_MANUAL_CORRESPONDENCE_ADDRESS, {
+        title: "What is the correspondence address?",
+        previousPage: "/sole-trader/correspondence-address--lookup",
```
Comment (sampankumar) [inline src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressManualController.ts:10]: These can go in the urls folder in pageURLs.ts

**Comment by swalke-ch** [comment]
Validation test should go in separate file similar to dateOfBirth.test.ts

**Comment by swalke-ch** [comment]
All new PRs expected to use localisation which is missing

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressManualController.ts (hunk 1 above)
Comment (sardhani-ch) [reply src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressManualController.ts]: addressed the comments

**Comment by sardhani-ch** [comment]
> All new PRs expected to use localisation which is missing

addressed the comments

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by sardhani-ch** [comment]
> Validation test should go in separate file similar to dateOfBirth.test.ts

My validations uses builtin express methods unlike dateOfBirth. Therefore written the tests in controller for validations.

File: src/main/routes/index.ts [hunk 2]
```diff
@@ -2,12 +2,13 @@ import { Router } from "express";
...
     indexController, soleTraderDateOfBirthController, soleTraderNameController,
     statementRelevantOfficerController, stopNotRelevantOfficerController, sectorYouWorkInController,
-    soleTraderCorrespondanceAddressManualController, soleTraderWhereDoYouLiveController,
+    soleTraderCorrespondenceAddressManualController, soleTraderWhereDoYouLiveController,
     soleTraderNationalityController, typeOfBusinessController, healthCheckController, OtherTypeOfBusinessController, soleTraderCorrespondenceAddressAutoLookupController,
     soleTraderCorrespodanceAddressDetailsController, soleTraderCorrespondenceAddressConfirmController
 
 } from "../controllers";
-import { correspondanceAddressManualValidator } from "../../../lib/validation/correspondanceAddressManual";
+// import { correspondenceAddressManualValidator } from "../../../lib/validation/correspondenceAddressManual";
```
Comment (swalke-ch) [inline src/main/routes/index.ts:10]: remove commented lines

File: src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressManualController.ts [hunk 3]
```diff
@@ -17,16 +22,17 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
...
+        const lang = selectLang(req.query.lang);
+        const locales = getLocalesService();
         const errorList = validationResult(req);
         if (!errorList.isEmpty()) {
-            const pageProperties = getPageProperties(formatValidationError(errorList.array()));
-            res.status(400).render(config.SOLE_TRADER_MANUAL_CORRESPONDANCE_ADDRESS, {
-                title: "What is your correspondence address?",
-                previousPage: "/sole-trader/correspondance-address--lookup",
-                pageProperties: pageProperties,
-                payload: req.body,
```
[the remark this answers is not in the document]
Comment (sardhani-ch) [reply src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressManualController.ts:27]: addressed

File: src/main/routes/index.ts (hunk 2 above)
Comment (sardhani-ch) [reply src/main/routes/index.ts]: 1. Addressed

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**